hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +62,106 @@
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static class LatencyInfo {
+        private Long totalLatency;
+        private Long operationCount;
+
+        LatencyInfo(Long totalLatency, Long operationCount) {
+            this.totalLatency = totalLatency;
+            this.operationCount = operationCount;
+        }
+
+        void add(Long totalLatency, Long operationCount) {
+            this.totalLatency += totalLatency;
+            this.operationCount += operationCount;
+        }
+
+        Long calculateAverage() {
+            return this.operationCount == 0 ? -1 : (this.totalLatency / this.operationCount);
+        }
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToGetLatency() {
+        return LocalMapStats::getTotalGetLatency;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToGetOperationCount() {
+        return LocalMapStats::getGetOperationCount;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToPutLatency() {
+        return LocalMapStats::getTotalPutLatency;
+    }
+
+    Function<? super LocalMapStats, ? extends Long> mapStatsToPutOperationCount() {
+        return LocalMapStats::getPutOperationCount;
+    }
+
+    private long mapOperationLatency(Node node, Predicate<MapConfig> predicate,
+                                     Function<? super LocalMapStats, ? extends Long> latencyMap,
+                                     Function<? super LocalMapStats, ? extends Long> operationMap) {
+        LatencyInfo latencyInfo = new LatencyInfo(0L, 0L);
+        mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
```

Comment:
there was a filter for `!= 0` here previously, afair. How do the tests even pass now? Don't we cover the division-by-zero case?